google/cloud/aiplatform/training_jobs.py

```diff
@@ -1753,3 +1754,329 @@ def _model_upload_fail_string(self) -> str:
+                Optional. The train budget of creating this Model, expressed in milli node
+                hours i.e. 1,000 value in this field means 1 node hour.
+                The training cost of the model will not exceed this budget. The final
+                cost will be attempted to be close to the budget, though may end up
+                being (even) noticeably smaller - at the backend's discretion. This
+                especially may happen when further model training ceases to provide
+                any improvements.
+                If the budget is set to a value known to be insufficient to train a
+                Model for the given training set, the training won't be attempted and
+                will error.
+                The minimum value is 1000 and the maximum is 72000.
+            model_display_name (str):
+                Optional. The display name of the managed AI Platform Model. The name
+                can be up to 128 characters long and can be consist of any UTF-8
+                characters. If a `base_model` was provided, the display_name in the
+                base_model will be overritten with this value. If not provided upon
+                creation, the job's display_name is used.
+            disable_early_stopping: bool = False
+                Required. If true, the entire budget is used. This disables the early stopping
+                feature. By default, the early stopping feature is enabled, which means
+                that training might stop before the entire training budget has been
+                used, if further training does no longer brings significant improvement
+                to the model.
+            sync: bool = True
+                Whether to execute this method synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
+        Returns:
+            model: The trained AI Platform Model resource or None if training did not
+                produce an AI Platform Model.
+
+        Raises:
+            RuntimeError: If Training job has already been run or is waiting to run.
+        """
+
+        if self._is_waiting_to_run():
+            raise RuntimeError("AutoML Image Training is already scheduled to run.")
+
+        if self._has_run:
+            raise RuntimeError("AutoML Image Training has already run.")
+
+        return self._run(
+            dataset=dataset,
+            training_fraction_split=training_fraction_split,
+            validation_fraction_split=validation_fraction_split,
+            test_fraction_split=test_fraction_split,
+            budget_milli_node_hours=budget_milli_node_hours,
+            model_display_name=model_display_name,
+            disable_early_stopping=disable_early_stopping,
+            sync=sync,
+        )
+
+    @base.optional_sync()
```

Comment:
We should force a dependency on `base_model` here to satisfy async workflows. Similar to how this method accepts a `dataset.Dataset` we should also pass in `base_model` here so it gets picked up by `base.optional_sync`.

Reply:
Good catch, added `base_model` as a param to `_run()`which is passed by `run()`